composer.json

```diff
@@ -4,7 +4,7 @@
     "license": "LGPL-3.0+",
     "description": "Extensible and flexible API for building web forms",
     "require": {
-        "neos/flow": "*"
+        "neos/flow": "~4.0 || dev-master"
```

Comment:
why always `~4.0 || dev-master` and not just `^4.0`? For development or testing the upgradeability, one has to set the required stability to `@dev` in the own root `composer.json`. Additionally, when the stable version is released, you won't have to update this again (since then you don't want to require `dev-master`).
Plus you could always require `dev-master as 4.0.x-dev` in your root `composer.json`.